spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SpringBootWebSecurityConfiguration.java

```diff
@@ -108,6 +108,15 @@ public static void configureHeaders(HeadersConfigurer<?> configurer,
 		if (!headers.isContentType()) {
 			configurer.contentTypeOptions().disable();
 		}
+		if (StringUtils.hasText(headers.getContentSecurityPolicy())) {
+			if (headers.getContentSecurityPolicyMode() == Headers.ContentSecurityPolicyMode.DEFAULT) {
+				configurer.contentSecurityPolicy(headers.getContentSecurityPolicy());
+			}
+			else {
+				assert headers.getContentSecurityPolicyMode() == Headers.ContentSecurityPolicyMode.REPORT_ONLY;
```

Comment:
Afraid not, we generally use Spring's `Assert` class. I can change this when we merge it.